### PR TITLE
Prevents some warnings from appearing in visual scripts

### PR DESCRIPTION
**Pull request by Chaosus**
Fixed warning 1:

vs_warning

Fixed warning 2:

vs_warning2

Fixed warning 3:

vs_warning3

Also added a check to prevent a node from connecting to itself:

image

### CODE REVIEW

**Review by AndreaCatania — APPROVED** [review]
Looks good!

File: modules/visual_script/visual_script_editor.cpp [hunk 1]
```diff
@@ -2836,6 +2836,23 @@ void VisualScriptEditor::_graph_connected(const String &p_from, int p_from_slot,
 
 	ERR_FAIL_COND(from_seq != to_seq);
 
+	if (from_seq) {
+		// checking to prevent warnings
+		if (script->has_sequence_connection(p_from.to_int(), from_port, p_to.to_int())) {
+			return;
+		}
+	} else {
```
Comment (fire) [inline modules/visual_script/visual_script_editor.cpp:2844]: Can you remove one layer of nested if statements?

**Review by fire — APPROVED** [review]
The comment is code style change. Looks good to me.

File: modules/visual_script/visual_script_editor.cpp (hunk 1 above)
Comment (Chaosus) [reply modules/visual_script/visual_script_editor.cpp]: Sure - Done

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Pull request merged**